### PR TITLE
Kevin/safely remove numbers

### PR DESCRIPTION
**Pull request by RocketKevin**
Nothing broke so far

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -101,10 +101,95 @@ class Board extends React.Component {
...
+    let backup = [];
+
+    //Change the random number if the position it's on was a zero
+    while(parseInt(valueBoard[randomPositionOfBoard]) === 0) {
+      //Reset random number
+      randomPositionOfBoard = Math.floor(Math.random() * valueBoard.length);
+    }
+
+    //Back up position of board's value removed and it's number
+    backup.push(randomPositionOfBoard);
```
Comment (brotatotes) [inline src/App.js:119]: Maybe you can do something like

`backup.push([randomPositionOfBoard, valueBoard[randomPositionOfBoard]])`

and then backup would be a bunch of size 2 lists, one for each pair.

File: src/App.js [hunk 2]
```diff
@@ -101,10 +101,95 @@ class Board extends React.Component {
...
+
+    //Set value of the random picked position to zero
+    valueBoard[randomPositionOfBoard] = 0;
+    
+    //Tell which position got their number removed and what number
+    return backup;
+  }
+
+  //Undo removeRandomNumber
+  undoRemoveRandomNumber(undo, valueBoard) {
```
Comment (brotatotes) [inline src/App.js:130]: it's not so clear what `undo` is here. Maybe update the name or add some more comments?

File: src/App.js [hunk 3]
```diff
@@ -101,10 +101,95 @@ class Board extends React.Component {
...
+    backup.push(valueBoard[randomPositionOfBoard]);
+
+    //Set value of the random picked position to zero
+    valueBoard[randomPositionOfBoard] = 0;
+    
+    //Tell which position got their number removed and what number
+    return backup;
+  }
+
+  //Undo removeRandomNumber
```
Comment (brotatotes) [inline src/App.js:129]: you don't need to have a comment if it's the same thing as the function name

File: src/App.js [hunk 4]
```diff
@@ -101,10 +101,95 @@ class Board extends React.Component {
...
+  undoRemoveRandomNumber(undo, valueBoard) {
+    //Reset the position, given from removeRandomNumber method, of the board's value
+    valueBoard[undo[0]] = undo[1];
+  }
+
+  findAllPossibleAnswers(valueBoard, backup) {
```
Comment (brotatotes) [inline src/App.js:135]: If I understand correctly, I think this is getting all the possible numbers for each backup square. But when I read the name I thought this function would try to find all possible solutions to the Sudoku puzzle. I wouldn't use the word `answers` here in the function name

File: src/App.js [hunk 5]
```diff
@@ -101,10 +101,95 @@ class Board extends React.Component {
...
+          oneSetOfAnswers.push(j);
+        }
+      }
+      allPossibleAnswers.push([backup[i][0], oneSetOfAnswers]);
+      oneSetOfAnswers = [];
+    }
+    return allPossibleAnswers;
+  }
+
+  /* In Progress */
```
Comment (brotatotes) [inline src/App.js:153]: Typically if something is in progress/not really working we don't want to include it in the PR.

File: src/App.js (hunk 4 above)
Comment (RocketKevin) [reply src/App.js:135]: Sorry for the bad names.....

File: src/App.js (hunk 5 above)
Comment (RocketKevin) [reply src/App.js:153]: Even if it doesn't affect the code?

File: src/App.js (hunk 1 above)
Comment (RocketKevin) [reply src/App.js:119]: It's already a bunch of size 2 lists. This: backup.push(this.removeRandomNumber(valueBoard)); | push them while grouping them up. You can test this out by logging backup in safetyRemoveNumbers method, inside the while loop.

Comment (brotatotes) [reply src/App.js:119]: Oh, I see. So in this funciton, `backup` means just one pair, while in the main thread `backup` means a list of all the pairs. Can you distinguish between these two names?

File: src/App.js (hunk 4 above)
Comment (brotatotes) [reply src/App.js:135]: haha no worries!

File: src/App.js (hunk 1 above)
Comment (RocketKevin) [reply src/App.js:119]: Sorry for the names...............

File: src/App.js (hunk 2 above)
Comment (RocketKevin) [reply src/App.js:130]: Done

File: src/App.js (hunk 3 above)
Comment (RocketKevin) [reply src/App.js]: Done

**Pull request merged**